plugins/api/discovery/src/main/java/org/apache/cloudstack/discovery/ApiDiscoveryServiceImpl.java

```diff
@@ -210,6 +211,24 @@ private ApiDiscoveryResponse getCmdRequestMap(Class<?> cmdClass, APICommand apiC
         return response;
     }
 
+    @Override
+    public List<String> listApiNames(Account account) {
+        List<String> apiNames = new ArrayList<>();
+        for (String apiName : s_apiNameDiscoveryResponseMap.keySet()) {
+            boolean isAllowed = true;
+            for (APIChecker apiChecker : _apiAccessCheckers) {
+                try {
+                    apiChecker.checkAccess(account, apiName);
+                } catch (Exception ex) {
+                    isAllowed = false;
```

Comment:
I think we won't need this variable - after the checkAccess method we can add the element to the list and on the catch block we can log the exception describing the error. Then the next if won't be needed either

Reply:
no, multiple access checkers can be called and any one may deny.